x/smart/psiphon_dialer_unimplemented.go

```diff
@@ -11,6 +11,8 @@ import (
 	"github.com/Jigsaw-Code/outline-sdk/transport"
 )
 
-func newPsiphonDialer(ctx context.Context, psiphonJSON []byte) (transport.StreamDialer, error) {
-	return nil, fmt.Errorf("To use psiphon configuration in x/smart the package must be built with the build tag `psiphon`: %w", errors.ErrUnsupported) 
+func newPsiphonDialer(ctx context.Context, psiphonJSON []byte, psiphonSignature string) (transport.StreamDialer, error) {
+	fmt.Printf("❌ Attempted to start psiphon tunnel %v but library was built without psiphon support. Please build using -tag psiphon.\n", psiphonSignature)
```

Comment:
Let's remove this. Let the caller decide what to print. The error already explains it.

Reply:
I think it would be worth keeping this for now. We're returning a more generic error to the client about all all fallbacks failing. Since there could be multiple errors for different individual fallbacks we don't attach them all to the returned error.

For example in example 2 in the PR text above the error we're returning to the client when the tag is missing is just the generic "Failed to find dialer: could not find a working fallback: all tests failed"

I am going to be more specific about our error behavior and exactly what we're returning in which failure cases when we have ✨more tests✨

Reply:
This is inconsistent with how we are logging things in the Smart Dialer though. You will have two messages with the Psiphon failure, no?

Reply:
Also, logging of strategies must use logCtx, or else we will emit spurious results after strategy selection is done.

Reply:
No, there won't be two messages. Currently we log only one message on a tag build failure. If we remove this line we'll log zero. See example 2 in "Tested" above.
I'm worried users will get stuck trying to debug their psiphon configs without realising they built the code wrong.

Agreed about the logging method though. I've converted both this and `psiphon_dialer.go` to pass in the StrategyFinder object and use logCtx. (Although in practice since the fallback selection happens in series, not in parallel, I haven't found late output to be an issue.)